### PR TITLE
Implement refresh displayed order labels

### PR DESCRIPTION
**Pull request by duncanspumpkin**
A surprisingly complex little bit of code for this.

### CODE REVIEW

File: src/OpenLoco/Windows/Vehicle.cpp [hunk 1]
```diff
@@ -2265,13 +2265,129 @@ namespace OpenLoco::Ui::Vehicle
...
+                    // The first line of the label will always be at the centre
+                    // of the station/waypoint. This works out where the subsequent
+                    // lines of the label will end up.
+                    auto width = (stringWidth + 3) << zoom;
+                    auto numberHeight = (unk.lineNumber * 10) << zoom;
+                    auto firstLineHeight = 11 << zoom;
+                    auto midX = width / 2;
+                    auto midFirstLineY = firstLineHeight / 2;
+
+                    unk.position.left[zoom] = (pos.x - midX) >> zoom;
```
Comment (marijnvdwerf) [inline src/OpenLoco/Windows/Vehicle.cpp:2384]: position -> frame. A position is a 2d coordinate.

I'd prefer making a rect first, and then only at the end do the zoom transformation, but that would be weird with the string width/number height already being in the final scale...

What about first determining the x/y in the map scale, and then only doing the positioning in endsize coordinates?

File: src/OpenLoco/Windows/Vehicle.cpp [hunk 2]
```diff
@@ -2265,13 +2265,129 @@ namespace OpenLoco::Ui::Vehicle
...
+                auto [loc, str] = sub_470B76(*order, i);
+                const auto pos = coordinate3dTo2d(loc.x, loc.y, loc.z, WindowManager::getCurrentRotation());
+                auto stringWidth = Gfx::getStringWidth(str);
+                for (auto zoom = 0; zoom < 4; ++zoom)
+                {
+                    // The first line of the label will always be at the centre
+                    // of the station/waypoint. This works out where the subsequent
+                    // lines of the label will end up.
+                    auto width = (stringWidth + 3) << zoom;
+                    auto numberHeight = (unk.lineNumber * 10) << zoom;
```
Comment (marijnvdwerf) [inline src/OpenLoco/Windows/Vehicle.cpp:2379]: 10  = lineheight constant

File: src/OpenLoco/Windows/Vehicle.cpp [hunk 3]
```diff
@@ -2265,13 +2265,129 @@ namespace OpenLoco::Ui::Vehicle
...
+                    for (auto innerOrder = orders.begin(); innerOrder->getOffset() != order.getOffset(); ++innerOrder)
+                    {
+                        if (rawOrder == innerOrder->getRaw())
+                        {
+                            lineNumber++;
+                        }
+                    }
+                }
+
+                _F2494A[i].lineNumber = lineNumber & 0xFF;
```
Comment (marijnvdwerf) [inline src/OpenLoco/Windows/Vehicle.cpp:2352]: Is the & 0xFF needed when assigning to an uint8?

Comment (duncanspumpkin) [reply src/OpenLoco/Windows/Vehicle.cpp:2352]: no but it was to draw attention to this stupidity of counting in a uint16 then saving as a uint8

File: src/OpenLoco/Windows/Vehicle.cpp (hunk 1 above)
Comment (duncanspumpkin) [reply src/OpenLoco/Windows/Vehicle.cpp]: I don't see a better way that would get the same results.

**Review by AaronVanGeffen — CHANGES_REQUESTED** [review]
No longer crashes, but the route numbers aren't showing up correctly. Basically, the number and cargo should be underneath the label, but they're displayed on top. (Probably a missing `+ lineHeight` somewhere?)

On master:
Screenshot

On this branch:
Screenshot (1)

**Comment by duncanspumpkin** [comment]
That should be it fixed. Took way to long to figure out the missing part of the calc.

**Review by AaronVanGeffen — APPROVED** [review]
Looks alright now :)

**Pull request merged**